http/http_connection.hpp

```diff
@@ -375,15 +375,36 @@ class Connection :
                               [self] { self->completeRequest(); });
         });
 
+        auto asyncResp = std::make_shared<bmcweb::AsyncResp>(res);
         if (thisReq.isUpgrade() &&
             boost::iequals(
                 thisReq.getHeaderValue(boost::beast::http::field::upgrade),
                 "websocket"))
         {
-            handler->handleUpgrade(thisReq, res, std::move(adaptor));
-            // delete lambda with self shared_ptr
-            // to enable connection destruction
-            res.setCompleteRequestHandler(nullptr);
+            asyncResp->res.setCompleteRequestHandler(
+                [self(shared_from_this())]() {
+                    if (self->res.resultInt() != 101)
+                    {
+                        // When any error occurs during handle upgradation,
+                        // the result in response will be set to respective
+                        // error. during success Result will be Switching
+                        // Protocols (101), which implies successful handle
+                        // upgrade. Response needs to be sent over this
+                        // connection only on failure.
+                        boost::asio::post(self->adaptor.get_executor(),
```

Comment:
This is still mostly unaddressed from upstream:
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/46989/1..6/http/http_connection.hpp#b374